susemanager-utils/susemanager-sls/src/modules/mgractionchains.py

```diff
@@ -178,27 +209,47 @@ def resume():
 
     This method is called by the Salt Reactor as a response to the 'minion/start/event'.
     '''
-    next_chunk = _read_next_ac_chunk()
-    if not next_chunk:
+    yaml_file = _read_next_ac_chunk()
+    if not yaml_file:
         return {}
-    if type(next_chunk) != dict:
+    if type(yaml_file) != dict:
         err_str = "Not able to resume Action Chain execution! Malformed " \
-                  "'_mgractionchains.conf' found: {0}".format(next_chunk)
+                  "'_mgractionchains.conf' found: {0}".format(yaml_file)
         log.error(err_str)
         raise CommandExecutionError(err_str)
-    next_chunk = next_chunk.get('next_chunk')
+    next_chunk = yaml_file.get('next_chunk')
     log.debug("Resuming execution of SUSE Manager Action Chain -> Target SLS: "
               "{0}".format(next_chunk))
-
-    inside_transaction = os.environ.get("TRANSACTIONAL_UPDATE")
-    if __grains__.get("transactional") and not inside_transaction:
-        return __salt__['transactional_update.sls'](next_chunk, queue=True)
+    
+    transactional_update = __grains__.get("transactional")
+    reboot_required = False
+    inside_transaction = False
+    if transactional_update:
+        reboot_required = yaml_file.get('reboot_required')
+        inside_transaction = os.environ.get("TRANSACTIONAL_UPDATE")
+
+    if transactional_update and reboot_required:
+        ret =  __salt__['transactional_update.sls'](next_chunk, queue=True, activate_transaction=True)
+    elif transactional_update and not inside_transaction:
+        ret =  __salt__['transactional_update.sls'](next_chunk, queue=True)
```

Comment:
Same as above, do they need to be seperate? Is it ok to be inside a transaction and call `__salt__['transactional_update.sls'](next_chunk, queue=True, activate_transaction=True)`?